src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+    var l50: Long = 0
+    while (0 <= i && blockLengthSum < fiftyPercent) {
+      blockLengthSum += countAndLengths(i)._1 * countAndLengths(i)._2
+      l50 += countAndLengths(i)._2
+      i -= 1
+    }
+
+    (n50, n90) match {
+      case (Some(a), Some(b)) => AssemblyStatistics(n50=a, n90=b, l50=l50)
+      case _ => unreachable("Could not compute n50/n90: " + counter.toList)
+    }
+  }
+}
+
+/**
+  * @param n50 the smallest block length such that the bases covered by all blocks this length and shorter are at least
+  * 50% of the # of bases covered by all blocks.
+  * @param n90 the smallest block length such that the bases covered by all blocks this length and shorter are at least
+  * 90% of the # of bases covered by all blocks.
+  * @param l50 the smallest number of blocks that whose sum produces the n50 value.
+  */
+private case class AssemblyStatistics(n50: Long, n90: Long, l50: Long)
+
+object PhaseBlockLengthMetric {
+  val MetricExtension = PathUtil.replaceExtension(Paths.get(AssessPhasingMetric.MetricExtension), ".block_lengths.txt").toString
+}
+
+/** Provides the number of phased blocks of a given length.
+  *
+  * @param is_truth true if the blocks are from the known/truth dataset, false otherwise
+  * @param length the length of the phased block
+  * @param count the number of phased blocks of the given length.
+  */
+case class PhaseBlockLengthMetric
+( is_truth: Boolean,
+  length: Long = 0,
+  count: Long = 0
+) extends Metric
+
+object PhasingUtils {
+  def getPhasingSetId(ctx: VariantContext): Int = {
+    ctx.getGenotype(0).getAttributeAsInt("PS", -1) // Integer.valueOf(ctx.getGenotype(0).getExtendedAttribute("PS", -1).toString)
+  }
+}
+
+private[vcf] class OverlapDetectorByContig(dict: SAMSequenceDictionary) {
+  import scala.collection.JavaConversions.collectionAsScalaIterable
+  private val detectors = dict.getSequences.map { seq =>
+    (seq.getSequenceName, new OverlapDetector[PhaseBlock](0, 0))
+  }.toMap
+
+  def addAll(iterator: Iterator[PhaseBlock]): this.type = {
+    iterator.foreach { block => this.detectors(block.getContig).addLhs(block, block) }
+    this
+  }
+
+  def overlapsAny(locatable: Locatable): Boolean = {
+    this.detectors(locatable.getContig).overlapsAny(locatable)
+  }
+
+  def getOverlaps(locatable: Locatable): Set[PhaseBlock] = {
+    import scala.collection.JavaConversions.asScalaSet
+    this.detectors(locatable.getContig).getOverlaps(locatable).toSet
+  }
+
+  def getAll: Iterator[PhaseBlock] = {
+    import scala.collection.JavaConversions.asScalaIterator
+    this.detectors.values.flatMap(_.getAll.iterator).toIterator
+  }
+}
+
+object PhaseBlock extends LazyLogging {
+  import scala.collection.mutable
+
+  /** Creates an overlap detector for blocks of phased variants.  Variants from the same block are found using the
+    * "PS" tag.
+    */
+  private[vcf] def toOverlapDetector(iterator: Iterator[VariantContext], dict: SAMSequenceDictionary): OverlapDetectorByContig = {
```

Comment:
`PhaseBlock.toOverlapDetector` makes me think it should be turning a `PhaseBlock` into an `OverlapDetector`.  How about `buildOverlapDetector`?